### PR TITLE
Remove networkPolicy/service dependency

### PR DESCRIPTION
**Pull request by astoycos**
Revert the changes made in #1921 Since the new `hairpin_snat_ip` option is now available in OVN 

The main objective here was to remove the dependency between network policy and services that was created in response to [this original bug](https://bugzilla.redhat.com/show_bug.cgi?id=1903651) which is 

When a POD sends traffic to a service it is loadbalanced(DNAT-ed) and sometimes the same pod is chosen as the backend to the service. To ensure this traffic travels through OVN-K8s rather than the POD2POD network since (srcIP==dstIP), the traffic is SNAT-ed to the VIP of the service's loadbalancer, which results in some packet loss with hairpinned traffic since this is not currently accounted for in the network Policy code

This solution simply forces the SNAT IP to be `ip4.src == 169.254.169.3 || ip6.src ==  fd69::3` in the hairpin case.  And also adds a default ingress allow ACL upon network policy creation to always allow such ingress traffic from the "special" hairpin service traffic.  It should always be allowed because in theory the only time an ingress packet will ever have a SRC-IP of  `ip4.src == 169.254.169.3 || ip6.src ==  fd69::3` is in the hairpin case

### CODE REVIEW

File: go-controller/pkg/ovn/gateway_init.go [hunk 1]
```diff
@@ -159,7 +159,8 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 				gatewayProtoLBMap[proto], stderr, err = util.RunOVNNbctl("--", "create",
 					"load_balancer",
 					fmt.Sprintf("external_ids:%s_lb_gateway_router=%s", proto, gatewayRouter),
-					fmt.Sprintf("protocol=%s", strings.ToLower(string(proto))))
+					fmt.Sprintf("protocol=%s", strings.ToLower(string(proto))),
+					"options:hairpin_snat_ip=169.254.169.3 fd69::3")
```
Comment (squeed) [inline go-controller/pkg/ovn/gateway_init.go:163]: Do we need to detect this feature?

File: go-controller/pkg/ovn/policy.go [hunk 2]
```diff
@@ -151,7 +148,11 @@ func getACLMatch(portGroupName, match string, policyType knet.PolicyType) string
 		aclMatch = "inport == @" + portGroupName
 	}
 
-	if match != "" {
+	if match == "hairpin" {
+		aclMatch += " && (ip4.src == 169.254.169.3 || ip6.src ==  fd69::3)"
```
Comment (squeed) [inline go-controller/pkg/ovn/policy.go:152]: Yeah.. these IPs need to be constants somewhere :). Otherwise they're pure magic.

Comment (squeed) [reply go-controller/pkg/ovn/policy.go:152]: also, add a docblock to this function that the match of "hairpin" is special constant.

File: go-controller/pkg/ovn/policy.go [hunk 3]
```diff
@@ -325,10 +335,11 @@ func (oc *Controller) createDefaultDenyPortGroup(ns string, nsInfo *namespaceInf
 }
 
 // modify ACL logging
-func (oc *Controller) setACLDenyLogging(ns string, nsInfo *namespaceInfo, aclLogging string) error {
+func (oc *Controller) setDefaultACLsLogging(ns string, nsInfo *namespaceInfo, aclLogging string) error {
```
Comment (squeed) [inline go-controller/pkg/ovn/policy.go:338]: This function needs an eventual rethink -- we shouldn't be reliant on the exact condition of the match; we should be setting ACLs with discoverable external_id's. But that's for another day...

File: go-controller/pkg/ovn/gateway_init.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/gateway_init.go:163]: I dont think we need to detect it (we always assume ovnk is on latest OVN), but we need to consider upgrade here and adding the option if it is missing on these LBs

File: go-controller/pkg/ovn/policy.go (hunk 3 above)
Comment (astoycos) [reply go-controller/pkg/ovn/policy.go:338]: I totally agree, I really dislike querying acl by match... External ID, name, or UUID would all be superior in my mind

File: go-controller/pkg/ovn/policy.go (hunk 2 above)
Comment (astoycos) [reply go-controller/pkg/ovn/policy.go:152]: Good Call will do!

File: go-controller/pkg/ovn/gateway_init.go (hunk 1 above)
Comment (astoycos) [reply go-controller/pkg/ovn/gateway_init.go:163]: @trozet So what's the best mechanism to do that? On ovnkube-master start should I look through all lbs?

**Comment by astoycos** [comment]
Waiting on #2236 to merge to implement upgrade logic

File: go-controller/pkg/ovn/gateway_init.go (hunk 1 above)
Comment (astoycos) [reply go-controller/pkg/ovn/gateway_init.go]: I make sure now that before creating any LBs we check if they already exist, then we update the options if needed

**Comment by astoycos** [comment]
@squeed @dcbw This should be pretty much ready if CI looks good

**Comment by astoycos** [comment]
Using #2321 to make sure things are looking good post upgrade

**Comment by astoycos** [comment]
It helps standardize how/ where we make Lbs across the codebase, and also provides an mechanism for ensuring that LBs are correctly configured in post upgrade scenarios.

**Comment by astoycos** [comment]
Hairpin test passes post upgrade on #2321 
```
• [SLOW TEST:5.322 seconds]
Service Hairpin SNAT
/home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/e2e.go:2644
  Should ensure service hairpin traffic is SNATed
  /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/e2e.go:2683
 ```

**Comment by astoycos** [comment]
CI is just canceling for some reason?

**Comment by astoycos** [comment]
@aojea I keep running into CI cancellations here, is there a way to see if the GH actions containers don't have enough resources?

**Pull request closed without merging**